collector/benthos/input/kubernetes.go

```diff
@@ -192,9 +202,35 @@ func (in *kubernetesResourcesInput) ReadBatch(ctx context.Context) (service.Mess
 			}
 
 			for _, pod := range podList.Items {
-				if !lo.EveryBy(pod.Status.ContainerStatuses, func(cs corev1.ContainerStatus) bool {
-					return cs.State.Running != nil
-				}) {
+				// Filter pods based on their status and configuration.
+				shouldInclude := false
+
+				switch pod.Status.Phase {
+				case corev1.PodRunning:
+					shouldInclude = true
+				case corev1.PodPending:
+					shouldInclude = in.includePendingPods
+				// Phase not set, check container statuses
```

Comment:
please extend comment to explain why `""` phase is possible

Reply:
It shouldn't be, but the documentation states:

> Most recently observed status of the pod. This data may not be up to date. Populated by the system.

I've also added logging if this happens.